crates/burn-jit/src/kernel/pool/standard_pool2d_shader.rs

```diff
@@ -0,0 +1,243 @@
+use std::marker::PhantomData;
+
+use crate::{
+    codegen::{Compilation, CompilationInfo, CompilationSettings, InputInfo, OutputInfo},
+    gpu::{gpu, Elem, Item, Scope, Variable, Visibility},
+    kernel::{DynamicKernelSource, SourceTemplate},
+    Compiler, JitElement, Runtime,
+};
+
+use super::PoolStrategy;
+
+pub(crate) struct StandardPool2dComputeShader<P: PoolStrategy, R: Runtime, E: JitElement> {
```

Comment:
Why Standard? Maybe add a comment listing other options (maybe future options).

Reply:
What I envisaged is that there would be standard vs adaptive shaders which relate to the layout/algorithm, with a pool strategy (max/max_with_indices/avg) which relate to what the computation does. 
But since we don't have adaptive_max_pooling I did not generalize the adaptive shader to the different pool strategies. Then since we don't really have several shader strategies, I dropped the standard terminology